src/project_variables.rs

```diff
@@ -3,28 +3,40 @@ use liquid::Object;
 use liquid_core::model::map::Entry;
 use liquid_core::Value;
 use regex::Regex;
+use serde::Deserialize;
 use thiserror::Error;
 
 use crate::config::{Config, TemplateSlotsTable};
 
-#[derive(Debug)]
+#[derive(Debug, Deserialize, Clone, PartialEq)]
 pub struct TemplateSlots {
     pub(crate) var_name: String,
     pub(crate) var_info: VarInfo,
     pub(crate) prompt: String,
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Deserialize, Clone, PartialEq)]
 pub enum VarInfo {
     Bool { default: Option<bool> },
     String { entry: Box<StringEntry> },
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Deserialize, Clone, PartialEq)]
 pub struct StringEntry {
     pub(crate) default: Option<String>,
     pub(crate) choices: Option<Vec<String>>,
-    pub(crate) regex: Option<Regex>,
+    pub(crate) regex: Option<String>,
```

Comment:
> could you please elaborate why Option<Regex> could not remain here?

The main reason is that `Regex` doesn't implement one or more of the new derive traits I added to `StringEntry`. But I also think it's not clean coding practice to store a type from another crate like this on a struct that maps to a config file data structure. 

> IMO keeping this as Regex we also would not need to construct it over and over when calling for validation.

I think you probably have better understanding of the code overall as well as more stake so I won't try to push my view here any further. Truthfully, it was more a matter of impulsivity and poor decision making with my time due to being sleepy that I even thought to contribute in the first place.

In retrospect, I probably should have stopped at opening the issue since I don't really have the time to commit to following through. If you don't think you can salvage anything here then it's probably best to close this so someone else can do it the right way.

Reply:
sorry to hear that. It was not my intent to discourage your PR efforts.

If you need any further support, you can also reach out to me on the matrix channel of the project at [![Chat on Matrix](https://img.shields.io/matrix/cargo-generate:matrix.org?server_fqdn=matrix.org)](https://matrix.to/#/#cargo-generate:matrix.org)